### PR TITLE
[Inspection Service] Add peer information and performance measurements

### PR DESCRIPTION
**Pull request by JoshLind**
Note: most of this PR is just tests.

### Description

This PR adds two new endpoints to the inspection service: (i) a simple `index` page (that lists all available endpoints); and (ii) a new `peer_information` endpoint that displays detailed network and peer information for each node. The purpose of the `peer_information` endpoint is to aid debugging and visibility for network and peer behaviors (including exposing the new monitoring metadata). See the example below for a preview.

The PR offers the following commits:
1. Add basic performance measurement information to the performance monitoring code (and tests!). This will allow the `peer_information` endpoint to display basic message throughput and bandwidth when running a performance benchmark.
2. Update the inspection service to handle the two new endpoints (including tests!).

Once this lands, I'll continue extending support for performance monitoring. 😄 It's still a work in progress.

### Test Plan
Existing and new unit tests. Also, when run locally (e.g., using a simple 2 node network), this is what the `peer_information` endpoint displays:
```
Peer information summary:
	- Number of peers: 1
	- Registered networks: [vfn, Validator]
	- Peers and network IDs: [Validator:2c7cff02]


Connection metadata for each peer:
	- Peer: Validator:2c7cff02, connection state: Connected, connection metadata: {"remote_peer_id":"2c7cff0231dd7818e6dc4f4671396b5a1ecb6e6005c5a8c4a36e900cff0d33a8","connection_id":0,"addr":"/ip4/127.0.0.1/tcp/53000/noise-ik/0xf0b0b25da306bd69a0e0ebd2003ceb117cc2db2b8392af2e2990c6b674bd8675/handshake/0","origin":"Inbound","messaging_protocol":"V1","application_protocols":{"inner":[231,184]},"role":"Validator"}


Trusted peers:
	- Network: vfn
	- Network: Validator
		- Peer: (2c7cff0231dd7818e6dc4f4671396b5a1ecb6e6005c5a8c4a36e900cff0d33a8, Peer { addresses: [/ip4/0.0.0.0/tcp/52922/noise-ik/0xf0b0b25da306bd69a0e0ebd2003ceb117cc2db2b8392af2e2990c6b674bd8675/handshake/0], keys: {x25519::PublicKey(f0b0b25da306bd69a0e0ebd2003ceb117cc2db2b8392af2e2990c6b674bd8675)}, role: validator })


Basic monitoring metadata for each peer:
	- Peer: Validator:2c7cff02, basic metadata: { average_ping_latency_secs: 0.0083719954, latest_network_info_response: { num_connected_peers: 1, distance_from_validators: 0 }, latest_node_info_response: { highest_synced_epoch: 2, highest_synced_version: 1163, ledger_timestamp_usecs: 1683081454550037,lowest_available_version: 0, uptime: 295.684601958s } }


Detailed monitoring metadata for each peer:
	- Peer: Validator:2c7cff02, detailed metadata: { average_ping_latency_secs: 0.0083719954, latest_network_info_response: NetworkInformationResponse { connected_peers: {Validator:02022504: ConnectionMetadata { network_address: /ip4/0.0.0.0/tcp/52936/noise-ik/0xe43d5f5fcdc4b91cf4ba556f24bede2b9ec5411e81f30560b6bb9d3cc42daf20/handshake/0, peer_id: 02022504395a6be76c70633ca233134d0b6791468504c40d8380faf2fec051ec, peer_role: validator }}, distance_from_validators: 0 }, latest_node_info_response: NodeInformationResponse { build_information: {"build_branch": "config_opt_6", "build_cargo_version": "cargo 1.66.1 (ad779e08b 2023-01-10)", "build_clean_checkout": "true", "build_commit_hash": "882f6ba5bbac963158e761d846c4e41f4c2c966f", "build_is_release_build": "false", "build_os": "macos-aarch64", "build_profile_name": "debug", "build_rust_channel": "1.66.1-aarch64-apple-darwin", "build_rust_version": "rustc 1.66.1 (90743e729 2023-01-10)", "build_tag": "", "build_time": "2023-05-02 22:21:31 -04:00", "build_using_tokio_unstable": "true"}, highest_synced_epoch: 2, highest_synced_version: 1163, ledger_timestamp_usecs: 1683081454550037, lowest_available_version: 0, uptime: 295.684601958s } }
...
```

### CODE REVIEW

**Review by brianolson — APPROVED** [review]

File: crates/inspection-service/src/inspection_service.rs [hunk 1]
```diff
@@ -243,6 +286,110 @@ pub fn start_inspection_service(node_config: NodeConfig) {
...
+
+/// Returns a simple text formatted string with peer and network information
+fn get_peer_information(peers_and_metadata: Arc<PeersAndMetadata>) -> String {
+    let mut peer_information = Vec::<String>::new();
+
+    // Display a summary of all peers and networks
+    let all_peers = peers_and_metadata.get_all_peers().unwrap_or_default();
+    let registered_networks: Vec<NetworkId> =
+        peers_and_metadata.get_registered_networks().collect();
+    peer_information.push("Peer information summary:".into());
```
[the remark this answers is not in the document]
Comment (JoshLind) [reply crates/inspection-service/src/inspection_service.rs:315]: Yeah, totally agree. This kind of hand rolled output formatting feels so ugly 😆

[the remark this answers is not in the document]
Comment (perryjrandall) [reply crates/inspection-service/src/inspection_service.rs:315]: Have we considered something minimal like tinytemplate?

https://docs.rs/tinytemplate/1.2.1/tinytemplate/

**Review by perryjrandall — APPROVED** [review]
Is it possible for these endpoints to be abused in some way?

File: crates/inspection-service/src/inspection_service.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoshLind) [reply crates/inspection-service/src/inspection_service.rs:315]: I haven't looked at anything specific thus far, but I'll definitely check it out 😄

**Comment by JoshLind** [comment]
> Is it possible for these endpoints to be abused in some way?

Good question. I don't believe so, e.g., we suggest folks always disable the inspection service from remote access. And even if someone could access it, they wouldn't see anything they couldn't already find (e.g., from the monitoring service on their own end). All in all, I think it's pretty harmless. 😄

**Pull request merged**